lib/streamlit/delta_generator.py

```diff
@@ -884,17 +898,24 @@ def _arrow_add_rows(
         return self
 
 
-DFT = TypeVar("DFT", bound=type_util.DataFrameCompatible)
-
-
-def _maybe_melt_data_for_add_rows(
```

Comment:
Since we now "melt" the data inside Vega-Lite, in the frontend, this function had to be updated.

NOTE: When using the legacy non-Arrow dataframe format we still "melt" in Python! Hence the "if arrow" in this function.